### PR TITLE
[Issue-79] Footer Links

### PR DESCRIPTION
**Pull request by NicaAlexandru**
https://github.com/aemsites/jet2holidays-portugal/issues/79

Test URLs:
- Before: https://main--jet2holidays-portugal--aemsites.aem.page/drafts/anica/footer-links
- After: https://issue-79--jet2holidays-portugal--aemsites.aem.page/drafts/anica/footer-links

Mobile (collapsed)

image

Mobile (expanded)

image

Desktop

image

### CODE REVIEW

File: blocks/footer-links/footer-links.css [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            padding-left: 24px;
+        }
+
+        .footer-links-grid-section > li:not(:last-child) {
+            padding-right: 24px;
+        }
+    }
+}
+
+@media (width >= 1200px) {
```
Comment (bstopp) [inline blocks/footer-links/footer-links.css:118]: nitpick but, use `min-width`

File: blocks/footer-links/footer-links.css [hunk 2]
```diff
@@ -0,0 +1,129 @@
+/* mobile first */
+.footer-links {
+    margin: 0 auto;
+    padding: 24px 27px;
+    background-color: #2b6fb1;
```
Comment (bstopp) [inline blocks/footer-links/footer-links.css:5]: Let's start pulling colors into `styles.css` with variable names.

File: blocks/footer-links/footer-links.css [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+    .mobile-heading {
+        display: flex;
+        flex-direction: row;
+        justify-content: space-between;
+        width: 100%;
+    }
+
+    .expand-icon,
+    .collapse-icon {
```
Comment (bstopp) [inline blocks/footer-links/footer-links.css:17]: you could also just use a `.open` additional class that is added when it's open. that way you don't have to swap the css.

File: blocks/footer-links/footer-links.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
+import {
+  domEl, h6, h5,
+} from '../../scripts/dom-helpers.js';
+
+const changeSectionClass = (oldClass, newClass) => {
+  const elements = document.querySelectorAll(`.${oldClass}`);
```
Comment (bstopp) [inline blocks/footer-links/footer-links.js:6]: See note about just adding a `.open` class.

**Review by bstopp — CHANGES_REQUESTED** [review]
I see what you did here, but may I make a suggestion.

Rather than swapping a bunch of class names around, When the  user clicks on the chevron to open the menu on mobile. you can just add an `.open` class to some parent container.

That class will add qualifiers that allow you to override the visibility of the list, and rotate the chevron as needed. 

This way you also don't have to do any kind of window resize check. A simple update to the visibility at the desired breakpoint when everything is visible, can be done - no JS needed.

File: blocks/footer-links/footer-links.css (hunk 1 above)
Comment (NicaAlexandru) [reply blocks/footer-links/footer-links.css:118]: I tried doing this, but I get the linting error:

`Expected "context" media feature range notation  media-feature-range-notation`

Maybe we can leave it as it is, or update the linting rules

**Comment by NicaAlexandru** [comment]
> I see what you did here, but may I make a suggestion.
> 
> Rather than swapping a bunch of class names around, When the user clicks on the chevron to open the menu on mobile. you can just add an `.open` class to some parent container.
> 
> That class will add qualifiers that allow you to override the visibility of the list, and rotate the chevron as needed.
> 
> This way you also don't have to do any kind of window resize check. A simple update to the visibility at the desired breakpoint when everything is visible, can be done - no JS needed.

**Comment by NicaAlexandru** [comment]
Hi @bstopp. I implemented your recommendations and included the .open logic, instead of the class swaps. 

However, I wanted to mention that we still need to check the window resize because the layout changes from mobile vs desktop. When shifting from desktop -> mobile, the main section collapses and when it goes from mobile -> desktop, it expands. As a result, depending on the window size, I add or remove the .open class

**Comment by bstopp** [comment]
> Hi @bstopp. I implemented your recommendations and included the .open logic, instead of the class swaps.
> 
> However, I wanted to mention that we still need to check the window resize because the layout changes from mobile vs desktop. When shifting from desktop -> mobile, the main section collapses and when it goes from mobile -> desktop, it expands. As a result, depending on the window size, I add or remove the .open class

Right - but what you can do is that the JS that handles the click adds the open class. 

But for desktop - you just use the `min-width` and override the individual values making the content hidden at that screen size. This removes the need for any resize monitoring logic.

**Comment by NicaAlexandru** [comment]
> > Hi @bstopp. I implemented your recommendations and included the .open logic, instead of the class swaps.
> > However, I wanted to mention that we still need to check the window resize because the layout changes from mobile vs desktop. When shifting from desktop -> mobile, the main section collapses and when it goes from mobile -> desktop, it expands. As a result, depending on the window size, I add or remove the .open class
> 
> Right - but what you can do is that the JS that handles the click adds the open class.
> 
> But for desktop - you just use the `min-width` and override the individual values making the content hidden at that screen size. This removes the need for any resize monitoring logic.

You are right, this is what I did: https://github.com/aemsites/jet2holidays-portugal/pull/80/files#diff-e6033faba034a50e47c8851257554a284f919d114e6bca11782036b0e5bf8046R14

**Comment by bstopp** [comment]
Here's a patch that shows how you can achieve the expanding, without requiring a monitor of the browser being resized. It handles showing/hiding the links at the specified breakpoint:

https://github.com/aemsites/jet2holidays-portugal/compare/issue-79...issue-79-upd

File: blocks/featured-resort-cards/featured-resort-cards.css [hunk 5]
```diff
@@ -102,7 +102,7 @@
     }
 }
 
-@media (width <= 900px) {
+@media (min-width: 900px) {
```
Comment (bstopp) [inline blocks/featured-resort-cards/featured-resort-cards.css:105]: This has to come after the 600px min-width, or it won't take effect.

**Review by bstopp — CHANGES_REQUESTED** [review]
Just need to move one thing, then it's good.

**Review by bstopp — APPROVED** [review]

**Pull request merged**